### ISSUE TITLE
[bug] MSBuildDeps incorrectly tries to import .props for non-transitive header-only packages that are not generated

### ISSUE THREAD

**Issue by RazielXYZ** [conan-io/conan#15624]
### Environment details

* Operating System+version: Windows 11
* Compiler+version: MSVC 17.8.6
* Conan version: 2.0.17
* Python version: 3.12.1

### Steps to reproduce

Create a conanfile for a MSBuild project as such:
```py
from conan import ConanFile
from conan.tools.microsoft import MSBuildDeps

class Pkg(ConanFile):
    settings = "os", "compiler", "arch", "build_type"
    
    def requirements(self):
        self.requires("scnlib/2.0.0")
     
    def generate(self):
        ms = MSBuildDeps(self)
        ms.generate()
```
Call conan install on the project with any msvc profile and output dir of your choice.
Look at the .props files written out to the output dir by conan MSBuildDeps: for example, for x64 release, in the file `conan_scnlib__scnlib_release_x64.props` there is a line `<Import Condition="'$(conan_fast_float_props_imported)' != 'True'" Project="conan_fast_float.props"/>`, but conan_fast_float.props does not exist.
This causes msbuild to fail when trying to build the project that imports the main conandeps.props generated by MSBuildDeps with `1>C:\testProj\conanprops\conan_scnlib__scnlib_release_x64.props(4,5): error MSB4019: The imported project "C:\testProj\conanprops\conan_fast_float.props" was not found. Confirm that the expression in the Import declaration "conan_fast_float.props" is correct, and that the file exists on disk.`
This is the same issue as here https://github.com/conan-io/conan/issues/15334 (and probably related to https://github.com/conan-io/conan/issues/14014) but is much easier/quicker to see with scnlib rather than opencv (which does exhibit the same issue in my case as well).

### Logs

conan install:
```
conan install . --output-folder=conanprops -pr:h=msvc193-x64-release -pr:b=msvc193-x64-release -vtrace

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=static
compiler.runtime_type=Release
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=static
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
Graph root
    conanfile.py: C:\testProj\conanfile.py
Requirements
    fast_float/6.1.0#85dc5bf9d3d6e605abf536818397baf3 - Cache
    scnlib/2.0.0#1001c947d4b617948955f7bd91dfe816 - Cache
    simdutf/4.0.5#484ca7bbfbbb2bd3db6289011f5ba613 - Cache
Build requirements
    cmake/3.28.1#92f79424d7b65b12a84a2180866c3a78 - Cache
Resolved version ranges
    cmake/[>=3.16 <4]: cmake/3.28.1

======== Computing necessary packages ========
Requirements
    fast_float/6.1.0#85dc5bf9d3d6e605abf536818397baf3:da39a3ee5e6b4b0d3255bfef95601890afd80709#69f6c6968a0f2f98bdffcadd07a3ad2d - Skip
    scnlib/2.0.0#1001c947d4b617948955f7bd91dfe816:260caab82f7f9a9b15225e54c089dc0403ae11eb#9732b7f31ad0ba2745ed3e082b277882 - Cache
    simdutf/4.0.5#484ca7bbfbbb2bd3db6289011f5ba613:a2e7df6f7bd606b2763e544f3a2c87dd6c478874#55f6770eed0b83432ab48f06caef5454 - Cache
Build requirements
    cmake/3.28.1#92f79424d7b65b12a84a2180866c3a78:522dcea5982a3f8a5b624c16477e47195da2f84f#29e6903b5006c2b2f81003c9ebf7551b - Skip

======== Installing packages ========
simdutf/4.0.5: Already installed! (1 of 2)
scnlib/2.0.0: Already installed! (2 of 2)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: scnlib/2.0.0

======== Finalizing install (deploy, generators) ========
conanfile.py: Calling generate()
conanfile.py: Generators folder: C:\testProj\conanprops
conanfile.py: Generating aggregated env files
conanfile.py: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
Install finished successfully
```
generated files:
```
ls .\conanprops\

        Directory: C:\testProj\conanprops


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a---        2024-02-08     04:39           2215   conan_scnlib__scnlib_release_x64.props
-a---        2024-02-08     04:39           1126   conan_scnlib__scnlib_vars_release_x64.props
-a---        2024-02-08     04:39            425   conan_scnlib__scnlib.props
-a---        2024-02-08     04:39            389   conan_scnlib.props
-a---        2024-02-08     04:39           1925   conan_simdutf_release_x64.props
-a---        2024-02-08     04:39            924   conan_simdutf_vars_release_x64.props
-a---        2024-02-08     04:39            404   conan_simdutf.props
-a---        2024-02-08     04:39             56   conanbuild.bat
-a---        2024-02-08     04:39            593   conanbuildenv-release-x86_64.bat
-a---        2024-02-08     04:39            272   conandeps.props
-a---        2024-02-08     04:39             54   conanrun.bat
-a---        2024-02-08     04:39            585   conanrunenv-release-x86_64.bat
-a---        2024-02-08     04:39             67   deactivate_conanbuild.bat
-a---        2024-02-08     04:39             65   deactivate_conanrun.bat
```

**Comment by memsharded** [comment]
Thanks very much for your report @RazielXYZ

It really  helped to identify the issue quickly and I am providing a fix in https://github.com/conan-io/conan/pull/15626 for next release (early next week). 

Having a way smaller repro case helped a lot to identify the issue and realize it was because of recipes using components, which I didn't manage to realize in the other ticket. This is really great for maintainers with a repo with huge activity, really appreciated.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15626 has been merged, it will be in next 2.1 (but if you want to give it a try from source before the release, you need to run from the ``develop2`` branch). Thanks again for reporting!
